src/plugins/intel_cpu/src/config.h

```diff
@@ -34,6 +35,7 @@ struct Config {
     std::string dumpToDot = "";
     int batchLimit = 0;
     size_t rtCacheCapacity = 5000ul;
+    std::shared_ptr<std::mutex> snippetMutex;
```

Comment:
Config is config. It assumes some plugin properties that can be adjusted. This is not right place the mutex. 
We already have mutex on ExecNetwork level that can be reused: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/exec_network.h#L56

Reply:
I agree with that `Config` is config so as we discussed I moved mutex from `Config` to` ExecGraph`.  I'd prefer to use separate mutex for snippets because I think these mutexes are for different cases: one for config and another for locking `clone_model` call in snippets ctor between streams.

Reply:
Don't agree. We have mutex on ExecNetwork level. We pass this mutex into Graphs as default synchronization primitive. From design perspective it can be used not only by Snippets. Lets keep things generic.

Reply:
Okey, I got it. I united these mutex and now we have common `shared_ptr` of `mutex`. It's more comfortable to move it to primitives